### PR TITLE
Added two Azure ML quickstart templates

### PR DESCRIPTION
**Pull request by denniseik**
Added two Azure ML quickstart templates for workspace creation @grayzu

### CODE REVIEW

File: quickstart/101-machine-learning/main.tf [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+}
+
+provider "azurerm" {
+  features {}
+}
+
+data "azurerm_client_config" "current" {}
+
+resource "azurerm_resource_group" "default" {
+  name     = "${var.name}-${var.environment}-rgp"
```
Comment (grayzu) [inline quickstart/101-machine-learning/main.tf]: To align with [Azure naming recommendations](https://docs.microsoft.com/en-us/azure/cloud-adoption-framework/ready/azure-best-practices/resource-abbreviations), this should be 
``` terraform
    name         = "rg-${var.name}-${var.environment}"
```

File: quickstart/101-machine-learning/main.tf [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+  location = "${var.location}"
```
Comment (grayzu) [inline quickstart/101-machine-learning/main.tf]: variables that are not concatenated do not need the string interpolation.
``` terraform
   location      = var.location
```

File: quickstart/101-machine-learning/readme.md [hunk 3]
```diff
@@ -0,0 +1,33 @@
+# Azure Machine Learning workspace
+
+This deployment template specifies an [Azure Machine Learning workspace](https://docs.microsoft.com/en-us/azure/machine-learning/concept-workspace),
```
Comment (grayzu) [inline quickstart/101-machine-learning/readme.md]: This is a bit of a nit, but in Terraform these are "configurations", not "templates".

File: quickstart/101-machine-learning/readme.md [hunk 4]
```diff
@@ -0,0 +1,33 @@
+# Azure Machine Learning workspace
+
+This deployment template specifies an [Azure Machine Learning workspace](https://docs.microsoft.com/en-us/azure/machine-learning/concept-workspace), 
+and its associated resources including Azure Key Vault, Azure Storage, Azure Application Insights and Azure Container Registry.
+
+This template describes the minimal set of resources you require to get started with Azure Machine Learning.
```
Comment (grayzu) [inline quickstart/101-machine-learning/readme.md]: same comment regarding "template" as above.

File: quickstart/101-machine-learning/variables.tf [hunk 5]
```diff
@@ -0,0 +1,17 @@
+variable "name" {
+  type        = string
+  description = "Name of the deployment"
+  default     = "azureml999"
```
Comment (grayzu) [inline quickstart/101-machine-learning/variables.tf]: This should either not have a default or use the [random provider](https://registry.terraform.io/providers/hashicorp/random/latest/docs) to generate a random name.

File: quickstart/101-machine-learning/workspace.tf [hunk 6]
```diff
@@ -0,0 +1,57 @@
+# Dependent resources for Azure Machine Learning
+resource "azurerm_application_insights" "default" {
+  name                = "${var.name}-${var.environment}-ain"
```
Comment (grayzu) [inline quickstart/101-machine-learning/workspace.tf]: This should align with Azure naming conventions mentioned above and be a prefix of "appi-" instead.

File: quickstart/101-machine-learning/workspace.tf [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+  location            = azurerm_resource_group.default.location
+  resource_group_name = azurerm_resource_group.default.name
+  application_type    = "web"
+}
+
+resource "azurerm_key_vault" "default" {
+  name                     = "${var.name}${var.environment}kv"
```
Comment (grayzu) [inline quickstart/101-machine-learning/workspace.tf]: This should align with Azure naming conventions mentioned above and be a prefix of "kv-" instead.

File: quickstart/101-machine-learning/workspace.tf [hunk 8]
```diff
@@ -0,0 +1,57 @@
...
+  purge_protection_enabled = false
+  
+  network_acls {
+    default_action = "Deny"
+    bypass = "AzureServices"
+  }
+}
+
+resource "azurerm_storage_account" "default" {
+  name                     = "${var.name}${var.environment}sa"
```
Comment (grayzu) [inline quickstart/101-machine-learning/workspace.tf]: This should align with Azure naming conventions mentioned above and be a prefix of "st" instead.

File: quickstart/101-machine-learning/workspace.tf [hunk 9]
```diff
@@ -0,0 +1,57 @@
...
+  account_replication_type = "GRS"
+
+  network_rules {
+    default_action = "Deny"
+    bypass = ["AzureServices"]
+  }
+}
+
+resource "azurerm_container_registry" "default" {
+  name                     = "${var.name}${var.environment}cr"
```
Comment (grayzu) [inline quickstart/101-machine-learning/workspace.tf]: This should align with Azure naming conventions mentioned above and be a prefix instead.

File: quickstart/101-machine-learning/workspace.tf [hunk 10]
```diff
@@ -0,0 +1,57 @@
...
+  location                 = azurerm_resource_group.default.location
+  resource_group_name      = azurerm_resource_group.default.name
+  sku                      = "Premium"
+  admin_enabled            = true
+}
+
+# Machine Learning workspace
+resource "azurerm_machine_learning_workspace" "default" {
+  name                    = "${var.name}-${var.environment}-aml"
```
Comment (grayzu) [inline quickstart/101-machine-learning/workspace.tf]: This should align with Azure naming conventions mentioned above and be a prefix of "mlw-" instead.

File: quickstart/201-machine-learning-private/main.tf [hunk 11]
```diff
@@ -0,0 +1,21 @@
...
+}
+
+provider "azurerm" {
+  features {}
+}
+
+data "azurerm_client_config" "current" {}
+
+resource "azurerm_resource_group" "default" {
+  name     = "${var.name}-${var.environment}-rgp"
```
Comment (grayzu) [inline quickstart/201-machine-learning-private/main.tf]: This should align with Azure naming conventions mentioned above and be a prefix of "rg-" instead.

File: quickstart/201-machine-learning-private/main.tf [hunk 12]
```diff
@@ -0,0 +1,21 @@
...
+  location = "${var.location}"
```
Comment (grayzu) [inline quickstart/201-machine-learning-private/main.tf]: should just be 
``` terraform
   location = var.location
```

File: quickstart/201-machine-learning-private/network.tf [hunk 13]
```diff
@@ -0,0 +1,89 @@
+# Virtual Network
+resource "azurerm_virtual_network" "default" {
+  name                = "${var.name}-${var.environment}-vnet"
```
Comment (grayzu) [inline quickstart/201-machine-learning-private/network.tf]: This should align with Azure naming conventions mentioned above and be a prefix instead.

File: quickstart/201-machine-learning-private/variables.tf [hunk 14]
```diff
@@ -0,0 +1,29 @@
+variable "name" {
+  type        = string
+  description = "Name of the deployment"
+  default     = "azureml568"
```
Comment (grayzu) [inline quickstart/201-machine-learning-private/variables.tf]: same as above. This should have no default or use random provider.

File: quickstart/201-machine-learning-private/workspace.tf [hunk 15]
```diff
@@ -0,0 +1,157 @@
+# Dependent resources for the workspace
```
Comment (grayzu) [inline quickstart/201-machine-learning-private/workspace.tf]: Please take a look at the Azure abbreviations guidance from above and align.

**Review by grayzu — CHANGES_REQUESTED** [review]
Thank you, Dennis, for the PR!!!

Overall this looks fantastic. With a few minor naming tweaks and syntax things, this will be good to go.

File: quickstart/101-machine-learning/main.tf (hunk 2 above)
Comment (denniseik) [reply quickstart/101-machine-learning/main.tf]: Thanks!

File: quickstart/101-machine-learning/readme.md (hunk 3 above)
Comment (denniseik) [reply quickstart/101-machine-learning/readme.md]: Much better name actually, will update it and as well in our docs

**Pull request merged**